Neos.Flow/Classes/Mvc/Exception/StopActionException.php

```diff
@@ -19,11 +19,38 @@
  * exception and - depending on the "dispatched" status of the request - either
  * continues dispatching the request or returns control to the request handler.
  *
- * See the Action Controller's forward() and redirectToUri() methods for more information.
+ * See {@see SpecialResponsesSupport::throwStopActionWithResponse()}
+ * or {@see SpecialResponsesSupport::responseRedirectsToUri()} for more information.
+ *
+ * Other control flow exceptions: {@see ForwardException}
  *
  * @api
  */
-class StopActionException extends \Neos\Flow\Mvc\Exception
+final class StopActionException extends \Neos\Flow\Mvc\Exception
 {
-    public ActionResponse $response;
+    /**
+     * The response to be received by the MVC Dispatcher.
+     */
+    public readonly ActionResponse $response;
+
+    private function __construct(string $message, int $code, ?\Throwable $previous, ActionResponse $response)
+    {
+        parent::__construct($message, $code, $previous);
+        $this->response = $response;
+    }
+
+    /**
+     * @param ActionResponse $response The response to be received by the MVC Dispatcher.
+     * @param string $details Additional details just for this exception, in case it is logged (the regular exception message).
+     */
+    public static function createForResponse(ActionResponse $response, string $details): self
```

Comment:
or `createToReturnResponse` or `createWithResponseToBeReturned`?

And `createToHandleNextRequest` respectively?

Reply:
Maybe we can make the factories internal? Then it wouldnt matter that much ^^

Reply:
it IS the api to trigger this behavior though, so technically this should be API.

Reply:
yes i agree, so do you like the namings or any of my suggestion better?